Add tertiary action guideline to modal section (#1336)
Shopify/polaris-react: src/components/Modal/README.md

Modals should:

- Only be closed by clicking the `X` or `Cancel` button and not by clicking the backdrop outside the modal, which is a large touch target that could result in accidental presses. Modals require merchants to take an action and should prevent the merchant from accidentally closing the modal without completing the required task.

> - Avoid having more than two buttons (e.g. primary and secondary) at the bottom. This prevents complex or unclear hierachy, as well as crowding on mobile screens. Since modals are for focused tasks, they should have focused actions. In some cases however, a [tertiary action](#tertiary-actions) may be appropriate.

Reviewer: ```suggestion
- Avoid having more than two buttons (primary and secondary) at the bottom. This prevents unclear action hierarchy and crowding on mobile screens. Since modals are for focused tasks, they should have focused actions. In some cases however, a [tertiary action](#tertiary-actions) may be appropriate.
```

Reviewer: we're trying to avoid latin abbreviations. normally we'd replace `e.g.` with for example, but this isn't an example, it's what the first two buttons are (if that makes sense), so I deleted the `e.g.`

<!-- end -->

<a name="tertiary-actions"></a>

### Tertiary actions

Tertiary actions should:

> - Only be used when the action requires the context of the content in the modal.

Reviewer: ```suggestion
- Only be used when the action requires the context of the content in the modal
```

Reviewer: removed period

> - Never dismiss the modal, unlike primary or secondary actions.

Reviewer: ```suggestion
- Never be used to dismiss the modal
```

Reviewer: simplified
